en/docs/developer-portal/manage-application.md

```diff
@@ -8,46 +8,150 @@ Let's get started!
-8. Click **:// CURL** to copy the cURL command template for generating the access token. You can add the consumer key and consumer secret to the URL and invoke the token endpoint to generate an API access token.
+8. Click **:// CURL** to copy the cURL command template for generating the access token. You can generate an API access token by adding the consumer key and consumer secret to the URL and invoking the token endpoint.
+
+### Renew an Access Token using the Refresh Token Grant Type
+
+!!! note
+    - To use this grant type, you will need the refresh token you recieved when you generated your current access token.
+    - Your application must have the **Refresh Token** grant type enabled.
+    - You can use a refresh token only once.
+
+You can use a refresh token to obtain a new access token when your current access token expires or when you need to renew the access token due to security reasons. You need to use the [Refresh Token grant type](#refresh-token-grant) for this purpose. 
+
+To generate a new access token and a new refresh token, you can invoke the Token API by replacing the `<refresh-token>` with the refresh token you received with the access token and replacing `<base64(consumerKey:consumerSecret)>` with the application's **Consumer Key** value and **Consumer Secret** value as follows:
+
+``` java tab="Format"
+curl -k -d "grant_type=refresh_token&refresh_token=<refresh-token>" -H "Authorization: Basic <base64(consumerKey:consumerSecret)>" -H "Content-Type: application/x-www-form-urlencoded" https://sts.choreo.dev/oauth2/token
+```
+
+``` java tab="Example"
+curl -k -d "grant_type=refresh_token&refresh_token=3154090c-37f1-3268-90f9-8bd84daf135c" -H "Authorization: Basic UXk3RUZfVEtMbEVLWTlVRFpiWHVscVA4ZVVBYTpKSWN3VTlIX1hGUFdTcW1RQmllZ3lJUzRKazhh" -H "Content-Type: application/x-www-form-urlencoded" https://sts.choreo.dev/oauth2/token
+```
+
+``` java tab="Response"
+{
+    "scope":"default",
+    "token_type":"Bearer",
+    "expires_in":3600,
+    "refresh_token":"7ed6bae2b1d76c041787e8c8e2d6cbf8",
+    "access_token":"b7882d23f1f8207f4bc6cf4a20633ab1"
+}
+```
+
+The above API response grants you a new access token and a refresh token.
+
+## Grant Types
+
+### Refresh Token Grant
+
+A refresh token is a token that you can obtain when your current access token is expired or when you need a new access token. You can use the refresh token grant type for this purpose. Issuing a refresh token is optional for the authorization server. If the authorization server issues a refresh token, it includes it in the response with the access token. You can use this refresh token and send it to the authorization server to obtain a new access token. Choreo's default authorization server, Asgardeo, issues refresh tokens for all other grant types other than the **client credentials** grant type, as recommended by the OAuth 2.0 specification.
```

Comment:
A refresh token is a token that you can use to get a new access token, when your current access token is expired.

Reply:
done

Reply:
This text doesn't look updated.